pliers/stimuli/vector.py

```diff
@@ -0,0 +1,109 @@
+import numpy as np
+import pandas as pd
+import requests
+import json
+from pliers.utils import attempt_to_import, verify_dependencies
+from pliers.stimuli.base import Stim
+
+
+class VectorStim(Stim):
+    ''' Vector stimulus (1-D numpy array)
+
+    Args:
+         array (np.ndarray, list or pd.Series): Vector of values. Can be list, 
+            array or pandas Series. Gets converted to 1-D numpy array.
+        labels (list): List of labels to which probability values refer, if 
+            that applies.
+        data_dict (dict): If defined, overwrites array and labels. Keys are 
+            passed to labels argument, values are passed to array argument.
+        filename (str): Path to file, if array has to be read from file. 
+            Can be a tsv file with vector values in one column, or json file
+            with labels as keys and array values as values.
+        data_column (str): If filename is defined, defines column to read
+            in as probability distribution
+        label_column (str): If filename is defined, defines columns where 
+            labels can be found.
+        onset (float): Optional onset of the event the probability 
+            distribution refers to.
+        duration (float): Optional duration of the event the probability 
+            distribution refers to.
+        order (int): Optional sequential index of the event the probability 
+            distribution refers to within some broader context.
+        name (str): Optional name to give to the Stim instance. If None is
+            provided, the name will be derived from the filename if one is
+            defined. If no filename is defined, name will be an empty string.
+        sort_data (str): 'descending' or 'ascending'. Sorts array (and labels)
+            in ascending or descending order.
+        url (str): Optional url to read contents from. Must be json readable
+            dictionary with labels as keys and values as probability values.
+    '''
+
+    _default_file_extension='.json'
+
+    def __init__(self, array=None, labels=None, filename=None, data_dict=None,
+        data_column=None, label_column=None, onset=None, duration=None, 
+        order=None, name=None, sort_data=None, url=None):
+
+        if filename is not None:
+            if filename.endswith('.json'):
```

Comment:
Sometimes filename endings are not sufficient to distinguish file type. Probably edge case though.